app/main.py

```diff
@@ -1,2 +1,146 @@
+from typing import Any, Hashable, NamedTuple
+
+
+DELETED = object()
+
+
+class Pair(NamedTuple):
+    key: Any
+    value: Any
+
+    def __str__(self) -> str:
+        return f"{self.key!r}: {self.value!r}"
+
+    def __repr__(self) -> str:
+        return f"({self.key!r}, {self.value!r})"
+
+
 class Dictionary:
-    pass
+    @classmethod
+    def from_dict(cls, dictionary: object, capacity: int = None) -> Any:
+        hash_table = cls(capacity or len(dictionary))
+        for key, value in dictionary.items():
+            hash_table[key] = value
+        return hash_table
+
+    def __init__(self,
+                 capacity: int = 8,
+                 load_factor_threshold: float = 0.6) -> None:
+        if capacity < 1:
+            raise ValueError("Capacity must be a positive")
+        if not (0 < load_factor_threshold <= 1):
+            raise ValueError("Load factor must be a number between (0, 1]")
+        self._slots = capacity * [None]
+        self._load_factor_threshold = load_factor_threshold
+
+    def __len__(self) -> int:
+        return len(self.pairs)
+
+    def __iter__(self) -> Any:
+        yield from self.keys
+
+    def __delitem__(self, key: Hashable) -> None:
+        for index, pair in self._probe(key):
+            if pair is None:
+                raise KeyError(key)
+            if pair is DELETED:
+                continue
+            if pair.key == key:
+                self._slots[index] = DELETED
+                break
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.load_factor >= self._load_factor_threshold:
+            self._resize_and_rehash()
+
+        for index, pair in self._probe(key):
+            if pair is DELETED:
+                continue
+            if pair is None or pair.key == key:
```

Comment:
```suggestion
            if pair is None or pair.key == key:
```
Also compare by hash

Reply:
In getitem the same

Reply:
No need to compare the hash.  Since the _probe function returns an element and an index that was calculated based on the hash (from key) and capacity (in function _index) .